packages/flutter/lib/src/cupertino/date_picker.dart

```diff
@@ -487,78 +487,75 @@ class CupertinoDatePicker extends StatefulWidget {
-        // just some dates are measured.
         for (int i = 1; i <= 12; i++) {
-          // An arbitrary date.
           final String date =
               localizations.datePickerMediumDate(DateTime(2018, i, 25));
-          if (longestText.length < date.length) {
-            longestText = date;
-          }
+          longTexts.add(date);
         }
       case _PickerColumnType.hour:
         for (int i = 0; i < 24; i++) {
           final String hour = localizations.datePickerHour(i);
-          if (longestText.length < hour.length) {
-            longestText = hour;
-          }
+          longTexts.add(hour);
         }
       case _PickerColumnType.minute:
         for (int i = 0; i < 60; i++) {
           final String minute = localizations.datePickerMinute(i);
-          if (longestText.length < minute.length) {
-            longestText = minute;
-          }
+          longTexts.add(minute);
         }
       case _PickerColumnType.dayPeriod:
-        longestText =
-          localizations.anteMeridiemAbbreviation.length > localizations.postMeridiemAbbreviation.length
-            ? localizations.anteMeridiemAbbreviation
-            : localizations.postMeridiemAbbreviation;
+        longTexts.add(localizations.anteMeridiemAbbreviation);
+        longTexts.add(localizations.postMeridiemAbbreviation);
       case _PickerColumnType.dayOfMonth:
         int longestDayOfMonth = 1;
-        for (int i = 1; i <=31; i++) {
+        for (int i = 1; i <= 31; i++) {
           final String dayOfMonth = localizations.datePickerDayOfMonth(i);
-          if (longestText.length < dayOfMonth.length) {
-            longestText = dayOfMonth;
-            longestDayOfMonth = i;
-          }
+          longTexts.add(dayOfMonth);
+          longestDayOfMonth = i;
         }
         if (showDayOfWeek) {
           for (int wd = 1; wd < DateTime.daysPerWeek; wd++) {
-            final String dayOfMonth = localizations.datePickerDayOfMonth(longestDayOfMonth, wd);
-            if (longestText.length < dayOfMonth.length) {
-              longestText = dayOfMonth;
-            }
+            final String dayOfMonth =
+                localizations.datePickerDayOfMonth(longestDayOfMonth, wd);
+            longTexts.add(dayOfMonth);
           }
         }
       case _PickerColumnType.month:
         for (int i = 1; i <= 12; i++) {
           final String month = standaloneMonth
               ? localizations.datePickerStandaloneMonth(i)
               : localizations.datePickerMonth(i);
-          if (longestText.length < month.length) {
-            longestText = month;
-          }
+          longTexts.add(month);
         }
       case _PickerColumnType.year:
-        longestText = localizations.datePickerYear(2018);
+        longTexts.add(localizations.datePickerYear(2018));
     }
 
-    assert(longestText != '', 'column type is not appropriate');
+    assert(longTexts.isNotEmpty, 'column type is not appropriate');
```

Comment:
I'm not sure this is equivalent to the previous logic. If empty strings are repeatedly added to the list, wouldn't the list be non-empty?

Reply:
Oh yes, indeed.